src/components/button/Button.tsx

```diff
@@ -46,18 +46,32 @@ export const Button: React.FC<ButtonProps> = ({
     `${loading ? 'loading' : ''}`,
     `${prefix}--${size}`
   );
-  if(variant==='destructive') {
+
+  const [width, setWidth] = React.useState(0);
+  const ref = React.useRef(null);
+  React.useEffect(() => {
+    if (ref.current && ref.current.getBoundingClientRect().width) {
```

Comment:
can't we do this with CSS only? 
What are we trying to achieve?

Reply:
To get dinamically the width of the button, this way when the button is loading it keep the same size

Reply:
To get dynamically the width of the button, this way when the button is loading it keep the same size

Reply:
I think that you can put the text to hidden, and center the loading icon with absolute

Reply:
done